src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1179,9 +1179,24 @@ private void getMethodSig(CORINFO_METHOD_STRUCT_* ftn, CORINFO_SIG_INFO* sig, CO
             Get_CORINFO_SIG_INFO(method, sig: sig, scope: null);
         }
 
-        private bool getMethodInfo(CORINFO_METHOD_STRUCT_* ftn, CORINFO_METHOD_INFO* info)
+        private bool getMethodInfo(CORINFO_METHOD_STRUCT_* ftn, CORINFO_METHOD_INFO* info, CORINFO_CONTEXT_STRUCT* context)
         {
             MethodDesc method = HandleToObject(ftn);
+
+            if (context != null && method.IsSharedByGenericInstantiations)
+            {
+                TypeSystemEntity ctx = entityFromContext(context);
+                if (ctx is MethodDesc methodFromCtx)
+                {
+                    method = methodFromCtx;
+                }
+                else if (ctx is InstantiatedType instantiatedCtxType)
+                {
+                    method = _compilation.TypeSystemContext.GetMethodForInstantiatedType(
```

Comment:
This won't work correctly for non-shared generic methods on shared generic types. 

For example, `G<string>.M<int>`. `GetTypicalMethodDefinition` will drop the method instantiation, so we will end up with `G<string>.M<T>`. I am not sure whether this problem is reachable in practice currently.

Reply:
I've just pushed a change:
```cs
MethodDesc methodTypicalDef = method.GetTypicalMethodDefinition();
if (methodTypicalDef.RequiresInstMethodDescArg())
{
    // Presumably, a rare case: we had MyClass<T1>.Foo<int>() method and
    // GetTypicalMethodDefinition() converted it to MyClass<T1>.Foo<T2>
}
else
{
    method = _compilation.TypeSystemContext.GetMethodForInstantiatedType(methodTypicalDef, instantiatedCtxType);
}
```
is it enough?